vnext/Microsoft.ReactNative/Views/FocusZoneViewManager.cpp

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+
+#include "FocusZoneViewManager.h"
+
+#include "ViewControl.h"
+
+#include "DynamicAutomationProperties.h"
+
+#include <Modules/NativeUIManager.h>
+#include <Utils/AccessibilityUtils.h>
+#include <Utils/Helpers.h>
+#include <Utils/PropertyUtils.h>
+
+#include <INativeUIManager.h>
+#include <IReactInstance.h>
+
+#include <winrt/Windows.System.h>
+#include <winrt/Windows.UI.Core.h>
+
+#include <UI.Xaml.Controls.Primitives.h>
+#include <UI.Xaml.Input.h>
+#include <winrt/Windows.ApplicationModel.Resources.Core.h>
+
+namespace winrt {
+using namespace Windows::Foundation;
+} // namespace winrt
+
+namespace Microsoft {
+namespace ReactNative {
+
+void FocusZoneViewShadowNode::createView() {
+  Super::createView();
+
+  auto panel = GetViewPanel();
+  m_panelKeyDownRevoker =
+      panel.PreviewKeyDown(winrt::auto_revoke, [=](const auto &sender, const auto &e) { OnKeyDown(sender, e); });
+  m_panelGettingFocusRevoker =
+      panel.GettingFocus(winrt::auto_revoke, [=](const auto &sender, const auto &e) { OnGettingFocus(sender, e); });
+  m_panelLosingFocusRevoker =
+      panel.LosingFocus(winrt::auto_revoke, [=](const auto &sender, const auto &e) { OnLosingFocus(sender, e); });
+}
+
+void FocusZoneViewShadowNode::updateProperties(winrt::Microsoft::ReactNative::JSValueObject &props) {
+  for (const auto &pair : props) {
+    const std::string &propertyName = pair.first;
+    const auto &propertyValue = pair.second;
+
+    if (propertyName == "focusDirection") {
+      if (propertyValue.Type() == winrt::Microsoft::ReactNative::JSValueType::String) {
+        m_isHorizontal = propertyValue.AsString() == "horizontal";
+      } else {
+        m_isHorizontal = false;
+      }
+    }
+  }
+}
```

Comment:
Sorry, I found another bug here. We need to call `Super::updateProperties(std::move(props));` so that View props are updated properly.